### DISCUSSION TITLE
View Tile3DLayer with OrbitView

### DISCUSSION THREAD

**Discussion by sidrubs** [visgl/deck.gl#7219] [Q&A]
We have 3D tiles hosted on Cesium ION which we are displaying using the Tile3DLayer. The default view for deck.gl is the MapView which works great. However, we have some use cases where we are not using a base map and the OrbitView behaviour would work better.

[This CodePen](https://codepen.io/srubsmodelar/pen/KKoLMpz) is an example Tile3DLayer loading well using MapView. However trying to achieve a similar thing with OrbitView does not seem to work (see [CodePen](https://codepen.io/srubsmodelar/pen/ExEzgjG)).

A couple notes on my OrbitView code:

- I am setting the Tile3DLayer's `coordinateSystem` to `COORDINATE_SYSTEM.CARTESIAN`.
- I have tried using the tileset `cartesianCenter` and `cartographicCenter` as the view state `target`, but neither shows the 3D tileset.

Is it possible to view a Tile3DLayer using OrbitView? Maybe the Tile3DLayer is being rendered, but I am looking in the incorrect location?

**Comment by Pessimistress** [comment]
I believe the 3D tiles loader expects the viewport to be geospatial (since you are rendering geospatial data). The `coordinateSystem` of the layer wouldn't matter here because the format of the coordinates is defined by the 3D Tiles specification.

What kind of controller behavior are you looking for?

**Reply by sidrubs** [reply]
When there is no base map present, we would like to be able to examine and zoom in to all angles of the 3D model. However, the closer the `pitch` is to 90 in MapView, the more erratic the zooming behaviour (when you add icons and text into the mix, things get really wild :sweat_smile: ). This makes sense because the map is the basis of all the controls and it is now perpendicular to the camera, so when you zoom your `latitude` and `longitude` values are going to change drastically.

As an example, see [this](https://codepen.io/srubsmodelar/pen/GRxawqz) CodePen. The pitch is set to `89` when the tileset loads (line 28), so zooming into the model is a bit erratic. If you set the pitch to exactly `90` one can't navigate at all.

If there was a method to make the MapView behave more like the OrbitView, that would be amazing.

**Reply by sidrubs** [reply]
@Pessimistress do you perhaps know of a method/settings to enable the MapView to allow one to examine and zoom in to all angles of the 3D model?

**Reply by Pessimistress** [reply]
`maxPitch` can be set to a value larger than 90, allowing you to view the bottom of the scene. The zoom behavior might be a bug.

**Reply by sidrubs** [reply]
Thank you for your response. Yeah, the zoom behavior gets a bit buggy. I'll have to give it some extra thought.

**Reply by mohadib** [reply]
@sidrubs did you ever have any luck using orbit view or making map view behave better? thanks

**Reply by sidrubs** [reply]
@mohadib sadly I never figured it out.

**Comment by charlieforward9** [comment]
how about 8 months later?

https://github.com/visgl/deck.gl/discussions/10014

Im planning a universal dgesture manager improvement that may interest you in scaling the level of control you could have over your OrbitView ? 👀🤞

Its not directly related - but reading into the rest of the system may help you better understand the fix.
